utils/nctl/sh/scenarios-upgrades/upgrade_scenario_01.sh

```diff
@@ -301,6 +301,25 @@ function _step_08()
             log "HASH MATCH :: $NODE_ID  :: HASH = $NX_STATE_ROOT_HASH :: N1 HASH = $N1_STATE_ROOT_HASH"
         fi
     done
+
+    log "Waiting for all nodes to sync to genesis"
+    TIMEOUT=300
+    for NODE_ID in $(seq 1 "$(get_count_of_nodes)")
+    do
+        log "Waiting for node $NODE_ID to sync to genesis (timeout=$TIMEOUT sec)"
+        RETRY_COUNT=1
+        LOWEST_AVAILABLE_BLOCK=$(get_node_lowest_available_block "$NODE_ID")
+        while [ -z "$LOWEST_AVAILABLE_BLOCK" ] || [ "$LOWEST_AVAILABLE_BLOCK" -ne 0 ]; do
+            if [ "$RETRY_COUNT" -gt $TIMEOUT ]; then
+                log "ERROR :: NODE-$NODE_ID RETRY :: Failed to sync to genesis within $TIMEOUT seconds; lowest available block $LOWEST_AVAILABLE_BLOCK"
+                exit 1
+            else
+                LOWEST_AVAILABLE_BLOCK=$(get_node_lowest_available_block "$NODE_ID")
+                sleep 1
+                ((RETRY_COUNT=RETRY_COUNT+1))
+            fi
+        done
+    done
```

Comment:
What do you think about extracting [do_await_node_1_historical_sync](https://github.com/casper-network/casper-node/blob/749cab93492a9f59c161c152732560e213511540/utils/nctl/sh/scenarios/upgrade_after_emergency_upgrade_test.sh#L200), change it to `do_await_node_historical_sync(node_num)` and use here?

Reply:
Actually, if you choose to do the above, you can then easily add this check to all of the upgrade tests, potentially saving us some trouble :)

Reply:
That's a good point, didn't see that we had that similar function. Will change.

Reply:
added in https://github.com/casper-network/casper-node/pull/3639/commits/e02e09277d030a9e81736eadb34d7e81aeb6aac9